uint8[34] data